src/ctx.cpp

```diff
@@ -133,7 +133,7 @@ zmq::ctx_t::~ctx_t ()
 
     //  If we've done any Curve encryption, we may have a file handle
     //  to /dev/urandom open that needs to be cleaned up.
-#ifdef ZMQ_HAVE_CURVE
+#if defined (ZMQ_USE_TWEETNACL)
```

Comment:
are you sure about this? sodium has this function too
https://download.libsodium.org/doc/generating_random_data/